cmd/minikube/cmd/start.go

```diff
@@ -1653,12 +1668,30 @@ func autoSetDriverOptions(cmd *cobra.Command, drvName string) (err error) {
 
 // validateKubernetesVersion ensures that the requested version is reasonable
 func validateKubernetesVersion(old *config.ClusterConfig) {
-	nvs, _ := semver.Make(strings.TrimPrefix(getKubernetesVersion(old), version.VersionPrefix))
+	kubernetesVer, err := getKubernetesVersion(old)
+	if err != nil {
+		paramVersion := viper.GetString(kubernetesVersion)
+		paramVersion = strings.TrimPrefix(strings.ToLower(paramVersion), version.VersionPrefix)
+		if errors.Is(err, ErrKubernetesPatchNotFound) {
+			exit.Message(reason.PatchNotFound, "Unable to detect the latest patch release for specified major.minor version v{{.majorminor}}",
+				out.V{"majorminor": paramVersion})
+		}
+		exit.Message(reason.Usage, `Unable to parse "{{.kubernetes_version}}": {{.error}}`, out.V{"kubernetes_version": paramVersion, "error": err})
+
+	}
+
+	nvs, _ := semver.Make(strings.TrimPrefix(kubernetesVer, version.VersionPrefix))
 	oldestVersion := semver.MustParse(strings.TrimPrefix(constants.OldestKubernetesVersion, version.VersionPrefix))
 	defaultVersion := semver.MustParse(strings.TrimPrefix(constants.DefaultKubernetesVersion, version.VersionPrefix))
 	newestVersion := semver.MustParse(strings.TrimPrefix(constants.NewestKubernetesVersion, version.VersionPrefix))
 	zeroVersion := semver.MustParse(strings.TrimPrefix(constants.NoKubernetesVersion, version.VersionPrefix))
 
+	paramVersion := viper.GetString(kubernetesVersion)
+	paramVersion = strings.TrimPrefix(strings.ToLower(paramVersion), version.VersionPrefix)
```

Comment:
I made the first change, but I don't think we can just replace line 1683.  kubernetesVer accounts for stable and latest as those checks/modifications are done in getKubernetesVersion. I think removing the use of getKubernetesVersion (kubernetesVer) here wouldn't make sense as that is the actual kubernetes version we have while the paramVersion can be something like 1.26 which we want so that we can tell the user their major.minor version not having a patch (as in the second before/after).

Reply:
Ah, I missed that it's using `kubernetesVer` instead of `paramVersion` so ignore the line 1683 suggestion, I think the rest of the comment still stands though